File: backend/app/api/faults.py
from flask import request, jsonify
from app.api import bp
from app.services.fault_service import FaultService

@bp.route('/faults', methods=['GET'])
def get_faults():
    product_model = request.args.get('product_model')
    keyword = request.args.get('keyword')
    
    if not product_model:
        return jsonify({'error': 'Missing product_model'}), 400
    
    if keyword:
        faults = FaultService.search_faults(product_model, keyword)
    else:
        faults = FaultService.get_faults_by_model(product_model)
    
    return jsonify({
        'faults': [fault.to_dict() for fault in faults]
    })

@bp.route('/faults/<int:fault_id>', methods=['GET'])
def get_fault_detail(fault_id):
    fault = FaultService.get_fault_detail(fault_id)
    if not fault:
        return jsonify({'error': 'Fault not found'}), 404
    return jsonify({'fault': fault.to_dict()})

@bp.route('/faults/<int:fault_id>/helpful', methods=['POST'])
def mark_helpful(fault_id):
    fault = FaultService.mark_helpful(fault_id)
    if not fault:
        return jsonify({'error': 'Fault not found'}), 404
    return jsonify({'message': 'Marked as helpful', 'fault': fault.to_dict()})

@bp.route('/faults/popular', methods=['GET'])
def get_popular_faults():
    limit = request.args.get('limit', 10, type=int)
    faults = FaultService.get_popular_faults(limit)
    return jsonify({
        'faults': [fault.to_dict() for fault in faults]
    })

@bp.route('/faults', methods=['POST'])
def create_fault():
    data = request.get_json()
    required_fields = ['product_model', 'fault_type', 'fault_desc', 'solution']
    
    for field in required_fields:
        if field not in data:
            return jsonify({'error': f'Missing {field}'}), 400
    
    fault = FaultService.create_fault(
        product_model=data['product_model'],
        fault_type=data['fault_type'],
        fault_desc=data['fault_desc'],
        solution=data['solution']
    )
    return jsonify({
        'message': 'Fault created',
        'fault': fault.to_dict()
    })

File: frontend/src/api/faults.js
import axios from 'axios'

const api = axios.create({
  baseURL: '/api',
  timeout: 10000
})

export function getFaults(productModel, keyword) {
  return api.get('/faults', {
    params: { product_model: productModel, keyword }
  })
}

export function getFaultDetail(faultId) {
  return api.get(`/faults/${faultId}`)
}

export function markHelpful(faultId) {
  return api.post(`/faults/${faultId}/helpful`)
}

export function getPopularFaults(limit = 10) {
  return api.get('/faults/popular', {
    params: { limit }
  })
}

export function createFault(data) {
  return api.post('/faults', data)
}

// ========== 重构 2026-09：分类→文件列表 ==========
export function getFaultCategories() {
  return api.get('/fault-categories')
}

export function getFaultsByCategory(categoryId) {
  return api.get('/faults', { params: { category_id: categoryId } })
}

// P0+P1 重构后，手机端 data source 从 fault.files 改为 fault.attachments
// 后端 CommonFault.to_dict() 默认已包含 attachments 列表，无需单独 endpoint
// 此函数保留供「按需刷新某故障的附件」场景使用
export function getFaultAttachments(faultId) {
  return api.get(`/admin/faults/${faultId}/attachments`)
}

// ========== 故障文件库 v2（2026-09 重构：移动端公开端点，无需登录） ==========

/**
 * 移动端获取公开的故障文件列表
 * @param {Object} params { kind?: 'pdf'|'image'|'doc' }
 */
export function getPublicFaultFiles(params = {}) {
  const p = {}
  if (params.kind) p.kind = params.kind
  return api.get('/fault-files', { params: p })
}